packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -574,7 +575,7 @@ export class Function extends FunctionBase {
   /**
    * The DLQ associated with this Lambda Function (this is an optional attribute).
    */
-  public readonly deadLetterQueue?: sqs.IQueue;
+  public readonly deadLetterQueue?: sqs.IQueue | sns.ITopic;
```

Comment:
Type unions are discouraged and should not be used, see [jsii documentation](https://aws.github.io/jsii/specification/2-type-system/#type-unions). I think we need to find another solution.

Maybe we should introduce a new property to send messages to a SNS topic: `readonly deadLetterTopic?: sns.ITopic;`
Either `deadLetterQueue ` or `deadLetterTopic` can be specified. If both properties are used, an error should be thrown.
@kaizen3031593 What do you think?

Reply:
Wasn't aware about that limitation. 
I like your idea but as usual, for me I mean, I don't like the ambiguity and a need to catch mentioned error that wouldn't be here with union.
On the other hand...if union does not work there is really no choice at all. 
I thought about doing discriminator kind of a interface but that is union all around once again.

Reply:
Let's wait until @kaizen3031593 confirms the proposal. I'm not sure if there are better ways to solve this.

Reply:
Thanks @jumic for linking to the jsii docs. We probably don't want a union type here, since such a thing does not exist in other jsii target languages.

I like the proposal of `deadLetterTopic` and maybe some additional documentation in `deadLetterQueue` pointing to the Topic option as an alternative.

Reply:
Fine by me. Will make necessary changes.